src/vstart.sh

```diff
@@ -575,18 +581,22 @@ EOF
 	fi
         if [ "$objectstore" == "bluestore" ]; then
             if [ "$spdk_enabled" -eq 1 ]; then
-                if [ "$(get_pci_selector)" == "" ]; then
+                if [ "$(get_pci_selector_num)" -eq 0 ]; then
                     echo "Not find the specified NVME device, please check."
                     exit
+                elif [ $(get_pci_selector_num) -lt $CEPH_NUM_OSD ]; then
+                    echo "OSD number is greater than NVME SSD number, please check."
+                    exit
+                else
+                    use_nvme_ssd=1
```

Comment:
i agree that there is chance the user could start `vstart.sh` with `--bluestore-spdk` but using a different objectstore backend. but can we assume that user alway want to use the bluestore if he/she pass `--bluestore-spdk` to `vstart.sh`? so we can dispense with yet another global variable and more assumptions that the SPDK backend always connects to an NVMe SSD? even we know that the SPDK backend is implemented using the spdk/nvme library. and under most circumstances, it will drive a SSD device. but i don't want to use this fact in the naming here.

Reply:
Thanks for comment！ I assume that if --bluestore-spdk is passed to vstart.sh, the user means he/she wants make the backend of all OSDs an NVME SSD. So if the number of OSD is greater than the number of available NVME SSDs, some of the OSDs get no NVME SSD to connect to. For example: when I run this command: sudo MON=1 MDS=0 MGR=1 RGW=0 OSD=2 ../src/vstart.sh -n -x -l -b -d --bluestore-spdk 8086:0953,
we have only one NVME SSD from 8086:0953,  but the OSD number is 2, so the first OSD is deployed correctly ,but when deploying the second one, there will be errors for the 2 OSDs can't use the same NVME SSD.

Reply:
@richael02 i think i understand your change here. i am just inclined to avoid adding `use_nvme_ssd` .

Reply:
@tchaikov  Thanks! I 'll overwrite spdk_enabled here .